File: hw/chavez/u3/hw17OPPConceptZoo/JavaScript/OPPConceptsZoo/model/Pig.js
import FarmAnimal from "./FarmAnimal.js";

export default class Pig extends FarmAnimal {

    constructor(id, breed, bornOn, weight) {
        super(id, breed, bornOn, weight);
    }

    cut() {
        return ["Ham", "Bacon", "Loin"];
    }

}

File: hw/tipan/u3/hw18OOPFarm/OPPConceptsZooJava/src/main/java/ec/edu/espe/oppconcetszoo/model/Pig.java
package ec.edu.espe.oppconcetszoo.model;

import ec.edu.espe.oppconcetszoo.controller.IMeatAnimal;
import java.util.ArrayList;

public class Pig extends FarmAnimal implements IMeatAnimal {

    public Pig(int id, String breed, Date bornOn, float weight) {
        super(id, breed, bornOn, weight);
    }

    @Override
    public void born() {
        System.out.println("A piglet was born.");
    }

    @Override
    public ArrayList<Cut> cut() {
        return new ArrayList<>();
    }

    @Override
    public void sendToSlaughterHouse(SlaughterHouse slaughterHouse) {
        System.out.println("Pig sent to slaughterhouse.");
    }

}